File: node/src/models/get-user-model.ts
import { db } from '../config/setup-database';
import { User } from '../types';

/** Gets the user from DB either by username or id */
export async function getUserFromDB(username:string){
    return new Promise<User|undefined>((resolve, reject)=>{
        db.get(`SELECT * FROM users WHERE id="${username}" OR username="${username}";`, (err, result)=>{
            if (err){
                reject(err);
            }
            return resolve(result);
        });
    });
}

File: go/models/getUserModel.go
package models

import (
	"api-battle/utils"
	"database/sql"
	"fmt"

	_ "github.com/mattn/go-sqlite3"
)

func GetUserFromDB(username string) (user utils.User, err error) {
	db, err := sql.Open("sqlite3", "../database.db")
	var newUser = new(utils.User)

	if err != nil {
		fmt.Println("🛑 Error openning database")
		return *newUser, err
	}
	stmt, err := db.Prepare("SELECT * FROM users WHERE id=? OR username=?;")
	if err != nil {
		fmt.Println("🛑 Error preparing query")
		return *newUser, err
	}

	res, err := stmt.Query(username, username)
	if err != nil {
		fmt.Println("🛑 Error executing query")
		return *newUser, err
	}
	res.Next()

	err = res.Scan(&newUser.Id, &newUser.Username, &newUser.Score)
	if err != nil {
		fmt.Println("🛑 Error scanning rows")
		return *newUser, err
	}
	res.Close()
	db.Close()
	return *newUser, err
}
